projects/packages/jetpack-mu-wpcom/src/features/media/heif-support.php

```diff
@@ -183,3 +183,29 @@ function jetpack_wpcom_add_heif_mimes_to_supported_upload_types( $mimes ) {
 	return $mimes;
 }
 add_filter( 'upload_mimes', 'jetpack_wpcom_add_heif_mimes_to_supported_upload_types' );
+
+/**
+ * Prevent WordPress from blocking HEIF/HEIC uploads in REST API.
+ *
+ * This allows HEIF files to pass through the REST API validation
+ * so they can be converted to JPEG by our conversion filter.
+ *
+ * @param bool        $check_mime Whether to check the mime type.
+ * @param string|null $mime_type          The mime type being uploaded.
+ * @return bool
+ */
+function jetpack_wpcom_allow_heif_uploads_in_rest_api( $check_mime, $mime_type ) {
+	if ( in_array( $mime_type, array( 'image/heif', 'image/heic' ), true ) ) {
+		return false;
+	}
+
+	return $check_mime;
+}
+
+/**
+ * Add the HEIF upload filter on REST API initialization.
```

Comment:
Does it mention to specify HEIF/HEIC for clarity?
```suggestion
 * Add the HEIF/HEIC upload filter on REST API initialization.
```